### PR TITLE
Support inkplate 5 and 5 V2

### LINKED ISSUE #2884: Add inkplate 5 display
**Describe the problem you have/What new integration you would like**

It would be nice to have support for the inkplate5 display. The Version for Inkplate 6v2 works well except for the screen dimensions. 

**Please describe your use case for this integration and alternatives you've tried:**

I tried the inkplate 6 v2 and it works, except that part of the screen is black due to the difference in screen resolution.

**Additional context**

see: https://soldered.com/product/soldered-inkplate-5-5-2%e2%80%b3-e-paper-board/

### PR DESCRIPTION
**Pull request by DavidSichau**
Provide support for inkplate 5 and 5 V2

# What does this implement/fix?

Waveform was used the one of the Inkplate repo.

Support inkplate 5 and 5V2 displays,

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes https://github.com/esphome/feature-requests/issues/2884

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#4276

## Test Environment

- [x] ESP32 Inkplate 5

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

### CODE REVIEW

File: esphome/components/inkplate6/inkplate.h [hunk 1]
```diff
@@ -64,7 +66,26 @@ class Inkplate6 : public display::DisplayBuffer, public i2c::I2CDevice {
...
+                                            // INKPLATE_5_V2
+                                            {{0, 0, 1, 1, 2, 1, 1, 1, 0},
+                                             {1, 1, 2, 2, 1, 2, 1, 1, 0},
+                                             {0, 1, 2, 2, 1, 1, 2, 1, 0},
+                                             {0, 0, 1, 1, 1, 1, 1, 2, 0},
+                                             {1, 2, 1, 2, 1, 1, 1, 2, 0},
+                                             {0, 1, 1, 1, 2, 0, 1, 2, 0},
+                                             {1, 1, 1, 2, 2, 2, 1, 2, 0},
+                                             {0, 0, 0, 0, 0, 0, 0, 0, 0}}
+};
```
Comment (jesserockz) [inline esphome/components/inkplate6/inkplate.h]:
```suggestion
                                             {0, 0, 0, 0, 0, 0, 0, 0, 0}}};
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Don't forget to add the models to the config options
https://github.com/esphome/esphome/blob/f652cd3851d17a4434876ab546021c6f698cb47f/esphome/components/inkplate6/display.py#L51-L56

**Comment by DavidSichau** [comment]
@jesserockz  I added the requested changes. Thanks for the review.

**Review by jesserockz — APPROVED** [review]

**Comment by DavidSichau** [comment]
Doc was added: https://github.com/esphome/esphome-docs/pull/4276

**Pull request merged**